File: examples/cpp/ui_demo.cpp
#include <iostream>
#include <string>
#include <vector>

#include <opencv2/highgui/highgui.hpp>

#include <vcp_ui/point_selection.h>
#include <vcp_ui/rect_selection.h>

#define VCP_LOG_LEVEL_DEBUG
#undef VCP_LOG_LOCATION
#include <vcp_utils/vcp_logging.h>

int main(int argc, char **argv)
{
  VCP_UNUSED_VAR(argc);
  VCP_UNUSED_VAR(argv);
  const cv::Mat image = cv::imread("flamingo.jpg");

  cv::Point point;
  if (vcp::ui::SelectPoint(point, image))
    VCP_LOG_INFO("Selected point: " << point);
  else
    VCP_LOG_INFO("No point selected");

  const auto points = vcp::ui::SelectPoints(image);
  if (points.empty())
  {
    VCP_LOG_INFO("No point selected");
  }
  else
  {
    for (const auto &pt : points)
      VCP_LOG_INFO("Selected: " << pt);
  }

  cv::Rect rect;
  if (vcp::ui::SelectRectangle(rect, image))
    VCP_LOG_INFO("Selected rect: " << rect);
  else
    VCP_LOG_INFO("No rect selected");

  return 0;
}

File: examples/python3/ui_demo.py
#!/usr/bin/env python
# coding=utf-8
"""Showcasing the simple UI tools of vcp - these are 
very basic user input tools. Consider using my 'iminspect'
python package (especially the image viewer and UI widgets)
instead."""

import os
import sys
import cv2


# Add path to the pypvt3 package
sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..', 'gen'))
from vcp import ui_basics
from vcp import imutils

if __name__ == "__main__":
    img = imutils.imread('../data/flamingo.jpg', flip_channels=True)

    # Select a rectangle
    valid, rect = ui_basics.select_rectangle(img, rect_color=(0, 255, 255), window_name='Rectangle selection demo')
    print('User confirmed: ', valid, ', rect: ', rect)

    # Select a single point
    valid, pt = ui_basics.select_point(img, point_color=(255, 0, 0), window_name='[Single] Point selection demo')
    print('User confirmed: ', valid, ', pt: ', pt)

    # Select multiple points
    pts = ui_basics.select_points(img, point_color=(255, 0, 0), window_name='[Multiple] Point selection demo')
    print('User selected pts: ', pts)